idd/Energy+.idd.in

```diff
@@ -96204,77 +96202,18 @@ LifeCycleCost:UsePriceEscalation,
-   N14, \field Year 13 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N15, \field Year 14 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N16, \field Year 15 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N17, \field Year 16 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N18, \field Year 17 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N19, \field Year 18 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N20, \field Year 19 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N21, \field Year 20 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N22, \field Year 21 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N23, \field Year 22 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N24, \field Year 23 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N25, \field Year 24 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N26, \field Year 25 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N27, \field Year 26 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N28, \field Year 27 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N29, \field Year 28 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N30, \field Year 29 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
-   N31; \field Year 30 Escalation
-        \type real
-        \note The escalation in price of the energy or water use for the year expressed as a decimal.
+   N11,N12,N13,N14,N15,N16,N17,N18,N19,N20,N21,N22,N23,N24,N25,N26,N27,N28,N29,N30, \field Year (n-1) Escalation
```

Comment:
Because it's not extensible past 100 years. Yes I agree no one should try exceeding 100 years, but to be safe I thought is was better to just fix the inputs.

Reply:
I also opened the defect file in the IDFEditor and got a lot of warnings of Too many fields for object. Anyone using more than 30 years will get these same warnings.

Reply:
You've done more than was needed here.  The changes to check `getObjectDefMaxArgs` would have been adequate. Extending idd objects like this is not preferred, but it's not worth pressing here.